File: java/hello_world/Usercode.java
// fq-class name com.corezoid.usercode.Usercode is mandatory
package com.corezoid.usercode;

import com.corezoid.gitcall.runner.api.UsercodeHandler;
import java.util.Map;

public class Usercode implements UsercodeHandler<Map<String, String>, Map<String, String>> {
    @java.lang.Override
    public Map<String, String> handle(Map<String, String> data) throws Exception {
        data.put("hello", "Hello world!");
        return data;
    }
}

File: go/user_error/usercode.go
package main

import (
	"context"
	"errors"

	"github.com/corezoid/gitcall-go-runner/gitcall"
)

func usercode(ctx context.Context, data map[string]any) error {
	_, ok := data["panic"]
	if ok {
		panic("my custom user panic")
	}
	return errors.New("my custom user error")
}
func main() {
	gitcall.Handle(usercode)
}
